tests/functional/RemoteTargetLocatorTest.php

```diff
@@ -71,4 +72,86 @@ public function testActiveElement()
         $this->assertSame('input', $activeElement->getTagName());
         $this->assertSame('test_name', $activeElement->getAttribute('name'));
     }
+
+    /**
+     * Data provider for frame().
+     */
+    public function frameProvider()
+    {
+        $parentFrame = 'This is the host page which contains an iFrame';
+        $childFrame = 'This is the content of the iFrame';
+        $otherChildFrame = 'open new window';
+
+        return [
+            'null' => [
+                null,
+                $parentFrame,
+            ],
+            'string 0' => [
+                '0',
+                $childFrame,
+            ],
+            'int 0' => [
+                0,
+                $childFrame,
+            ],
+            'float 0' => [
+                0.0,
+                $childFrame,
+            ],
+            'string 1' => [
+                '1',
+                $otherChildFrame,
+            ],
+            'int 1' => [
+                1,
+                $otherChildFrame,
+            ],
+            'floa 1' => [
+                1.0,
+                $otherChildFrame,
+            ],
```

Comment:
I would prefer having datasets formatted on one row if they fits, like this:
```php
 'null' => [null, $parentFrame],
```

Reply:
I've removed some of these test cases as they were invalid.
We should not accept string or floats and cast to int because the frame number comes from a Number and it is perfectly valid to have a frame whose name is a stringy int.

I've added new tests for stringy integers.

I also detected an existing bug with the JSONWire implementation where it did not support being passed null.
I've confirmed that the following are supported in JSONWire (https://github.com/SeleniumHQ/selenium/wiki/JsonWireProtocol#sessionsessionidframe):
> id - {string|number|null|WebElement JSON Object} Identifier for the frame to change focus to.